### PR TITLE
Add -failfast flag when the mode is fail_fast

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
Though there is mention of mode `fail_fast`, at https://github.com/etcd-io/etcd/blob/main/scripts/test_lib.sh#L252, and usages, there is no actual pass of `-failfast` flag.

_Usages locations:_
https://github.com/etcd-io/etcd/blob/main/scripts/test.sh#L163
https://github.com/etcd-io/etcd/blob/main/scripts/test.sh#L168

### CODE REVIEW

**Review by serathius — APPROVED** [review]

File: scripts/test_lib.sh
Comment (jmhbnz) [inline scripts/test_lib.sh:312]: I think this change introduces a subtle bug, but it is an easy fix.

On line 296 there is currently a final mention of `goTestFlags="-v"` which does not add any trailing space padding.  So if flags are both `verbose` and `fail_fast` I believe the resulting string will look like `-v-failfast` without any spaces inbetween.

Can you please double check this and adjust spacing to resolve if my theory is right?

**Review by jmhbnz — APPROVED** [review]

**Review by ptabor — APPROVED** [review]
I think that the role of this flag was to drive the behavior across many packages rather than pass it down to a specific package to not continue executing other tests in the package.

If I understand the tradeoff with this change is mostly about our pre-merge test: 
 - tests (if failing) will fail faster
 - the results will be less complete (as they will contain only a single failed test, rather than all of them in a single package).

As we expect developer to rerun the whole target locally after fixing a test it sounds like a fair tradeoff.

**Comment by Rajalakshmi-Girish** [comment]
Will this PR need an LGTM comment to merge? @jmhbnz Can you please help me understand what it needs to merge this one.

**Comment by Rajalakshmi-Girish** [comment]
> Will this PR need an LGTM comment to merge? @jmhbnz Can you please help me understand what it needs to merge this one.

or may be should all checks be successful?

**Comment by serathius** [comment]
Two maintainer approvals are enough.

**Comment by serathius** [comment]
Let's wait for test to pass.

**Comment by Rajalakshmi-Girish** [comment]
> Let's wait for test to pass.
Will it auto merge if all tests pass and there are two approvals?

**Pull request merged**
